engine/collection/compliance/core.go

```diff
@@ -81,17 +80,19 @@ func NewCore(
 
 // OnBlockProposal handles incoming block proposals.
 func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.ClusterBlockProposal) error {
-	header := proposal.Header
+	block := proposal.Block.ToInternal()
```

Comment:
I am confused why this transformation `ToInternal()` is repeatedly happening while processing a block proposal: 
* https://github.com/onflow/flow-go/blob/4d1829104f468fdbe6a1399b457031006511a267/engine/collection/compliance/core.go#L199
* https://github.com/onflow/flow-go/blob/4d1829104f468fdbe6a1399b457031006511a267/engine/collection/compliance/core.go#L249

In my head, the conversion would be done once, which created certain guarantees in the resulting data structure. For example here, `ToInternal()` gives rise to the higher-level internal representation with certain guarantees, e.g. the relevant pointers are not nil. 

I would suggest to convert _once_ right here in the beginning of `OnBlockProposal`. (Probably beyond the scope of this PR, but that is where I would imagine the pre-computation of the ID would happen: also as part of the `ToInternal()` function.) If we want to work with `cluster.Block` from here on, we would need to update `PendingClusterBlockBuffer` to work with the internal representation instead, which I think would be worth it.

Reply:
> In my head, the conversion would be done once, which created certain guarantees in the resulting data structure.

Good point. The multiple conversions (and usages) of the wire type is partly why I was unhappy with this approach - I think this will help to reduce the technical debt scope as well.